dtfs-central-api/src/services/state-machines/fee-record/event-handlers/helpers/get-fixed-fee-for-facility.ts

```diff
@@ -8,11 +8,20 @@ import { calculateFixedFee } from './calculate-fixed-fee';
  * @param facilityId - The (ukef) facility id
  * @param utilisation - The facility utilisation
  * @param reportPeriod - The report period
+ * @param coverPercentage - TFM facility cover percentage value
+ * @param coverEndDate - TFM facility cover end date
+ * @param interestPercentage - TFM facility interest percentage value
+ * @param dayCountBasis - TFM facility day count basis value
  * @returns The fixed fee for the supplied report period
  */
-export const getFixedFeeForFacility = async (facilityId: string, utilisation: number, reportPeriod: ReportPeriod) => {
-  const { coverEndDate, dayCountBasis, interestPercentage, coverPercentage } = await getLatestTfmFacilityValues(facilityId, reportPeriod);
-
+export const getFixedFeeForFacility = (
+  utilisation: number,
+  reportPeriod: ReportPeriod,
+  coverPercentage: number,
+  coverEndDate: Date,
+  interestPercentage: number,
+  dayCountBasis: number,
+) => {
   const ukefShareOfUtilisation = calculateUkefShareOfUtilisation(utilisation, coverPercentage);
```

Comment:
Please can we calculate ukefs share just the once inside `handleFeeRecordGenerateKeyingDataEvent` and pass the same value into the fixed fee calculations and the utilisation adjustment calculations rather than recalculating it here.

Reply:
done